### PR TITLE
boards: nrf52840_pca10056: Rename board to nrf52840dk_nrf52840

### PR DESCRIPTION
**Pull request by anangl**
The nRF52840 DK board target, so far known as nrf52840_pca10056, is renamed to nrf52840dk_nrf52840.
Its documentation and all references to its name in the tree are updated accordingly. Overlay and configuration files specific to this board are also renamed, to match the new board name.

In the same manner nrf52811_pca10056 (which in fact uses nRF52840 DK for emulation of nRF52811) is renamed to nrf52840dk_nrf52811.

This PR contains also a slight update of the nRF52840 DK documentation and its picture.

### CODE REVIEW

**Review by ioannisg — APPROVED** [review]

**Comment by henrikbrixandersen** [comment]
What is the motivation for renaming the board? I suspect a lot of users will be impacted by these renames.

**Comment by ioannisg** [comment]
> What is the motivation for renaming the board? I suspect a lot of users will be impacted by these renames.

@henrikbrixandersen these board definitions correspond to the official nRF DKs and the initiative to rename comes from Nordic :) However, to address your concern, some care has been taken to mitigate the effects of the migration; there's still ability to build using the old names (a Warning in CMake is issued.)
Please, see a related PR for nRF51 DK here: #https://github.com/zephyrproject-rtos/zephyr/pull/23447

You are welcome to provide feedback there!

**Comment by carlescufi** [comment]
@anangl please rebase and remove from Draft status

File: doc/conf.py [hunk 1]
```diff
@@ -340,7 +340,7 @@
 		('boards/arm/nrf51_pca10028/doc/nrf51_pca10028', 'boards/arm/nrf51_pca10028/doc/index'),
 		('boards/arm/nrf51_vbluno51/doc/nrf51_vbluno51', 'boards/arm/nrf51_vbluno51/doc/index'),
 		('boards/arm/nrf52810_pca10040/doc/nrf52810_pca10040', 'boards/arm/nrf52810_pca10040/doc/index'),
-		('boards/arm/nrf52840_pca10056/doc/nrf52840_pca10056', 'boards/arm/nrf52840_pca10056/doc/index'),
```
Comment (ioannisg) [inline doc/conf.py]: You need to add a new entry here, see how it was done in nRF51

**Review by ioannisg — CHANGES_REQUESTED** [review]
in boards/deprecated.cmake you need to add an entry for the old names of the nrf52840 boards

File: doc/conf.py (hunk 1 above)
Comment (anangl) [reply doc/conf.py]: Done.

**Comment by anangl** [comment]
- rebased on master
- added new entries to `boards/deprecated.cmake`
- corrected `doc/conf.py`

**Review by ioannisg — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Comment by carlescufi** [comment]
@anangl can you add something like https://github.com/zephyrproject-rtos/zephyr/commit/6251530f124725779dfab2e8eae2850f4eab2a30?

**Comment by anangl** [comment]
> @anangl can you add something like [6251530](https://github.com/zephyrproject-rtos/zephyr/commit/6251530f124725779dfab2e8eae2850f4eab2a30)?

@carlescufi It's already in place, see:
- [boards/arm/nrf52840dk_nrf52811/doc/index.rst](https://github.com/zephyrproject-rtos/zephyr/pull/23524/files#diff-d09b31181f71933c6112d2a08d59620fR18)
- [boards/arm/nrf52840dk_nrf52840/doc/index.rst ](https://github.com/zephyrproject-rtos/zephyr/pull/23524/files#diff-54ee31d835e82f72b7a2bbef15bce694R39)

**Comment by anangl** [comment]
Restored mentions of PCA10056 markings in documentation for both boards.

**Comment by anangl** [comment]
Reverted renaming of `nrf52840_pca10056.jpg` located in `boards/arm/nrf52840_pca10090/doc/img`, a file that is not related to the changes done in this PR (and probably should be removed, in a PR renaming the nrf52840_pca10090 board, as it does not seem to be appropriate for that board definition and it also does not seem to be used at all).

**Comment by ioannisg** [comment]
Could some more reviewers take a brief look here? We would like to finalize the review round in the PR. Thanks!

**Review by joerchan — APPROVED** [review]

**Review by vanwinkeljan — APPROVED** [review]
+1 for display related changes

**Review by henrikbrixandersen — APPROVED** [review]

**Pull request merged**